auctions/english-auction/tests/utils/mod.rs

```diff
@@ -0,0 +1,349 @@
+            auction: EnglishAuction::new(auction_id.to_string(), wallet4.clone()),
+            wallet: wallet4.clone(),
+        };
+
+        let sell_amount = 10;
+        let inital_price = 1;
+        let reserve_price = 10;
+        let time = 10;
+
+        (
+            deploy_wallet,
+            seller,
+            buyer1,
+            buyer2,
+            sell_asset_id,
+            buy_asset_id,
+            sell_amount,
+            inital_price,
+            reserve_price,
+            time,
+        )
+    }
+
+    pub async fn token_asset(contract_id: ContractId, amount: u64) -> Asset {
+        let token = TokenAsset {
+            contract_id,
+            amount,
+        };
+
+        Asset::TokenAsset(token)
+    }
+
+    pub async fn nft_asset(contract_id: ContractId, token_ids: u64) -> Asset {
+        let token = NFTAsset {
+            contract_id,
+            token_ids,
+        };
+
+        Asset::NFTAsset(token)
+    }
+}
+
+pub mod abi_calls {
+
+    use super::*;
+
+    pub async fn init_token(
+        deploy_wallet: &Metadata,
+        seller: &Metadata,
+        sell_asset_id: ContractId,
+        sell_amount: u64,
+        _buy_asset_id: ContractId,
+        inital_price: u64,
+        reserve_price: u64,
+        time: u64,
+        buy_asset_struct: Asset,
+        sell_asset_struct: Asset,
+    ) -> u64 {
+        deploy_funds(&deploy_wallet, &seller.wallet, 100).await;
+
+        let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+        let call_params =
+            CallParameters::new(Some(sell_amount), Some(AssetId::from(*sell_asset_id)), None);
+
+        seller
+            .auction
+            .constructor(
+                englishauction_mod::Identity::Address(seller.wallet.address()),
+                sell_asset_struct,
+                buy_asset_struct,
+                inital_price,
+                reserve_price,
+                time,
+            )
+            .tx_params(tx_params)
+            .call_params(call_params)
+            .call()
+            .await
+            .unwrap()
+            .value
+    }
+
+    pub async fn init_nft(
+        _deploy_wallet: &Metadata,
+        seller: &Metadata,
+        _sell_asset_id: ContractId,
+        _sell_amount: u64,
+        _sell_nft_id: u64,
+        _buy_asset_id: ContractId,
+        inital_price: u64,
+        reserve_price: u64,
+        time: u64,
+        buy_asset_struct: Asset,
+        sell_asset_struct: Asset,
+    ) -> u64 {
+        seller
+            .auction
+            .constructor(
+                englishauction_mod::Identity::Address(seller.wallet.address()),
+                sell_asset_struct,
+                buy_asset_struct,
+                inital_price,
+                reserve_price,
+                time,
+            )
+            .call()
+            .await
+            .unwrap()
+            .value
+    }
+
+    pub async fn bid_tokens(
```

Comment:
they are not abi functions. these are helpers similar to the setup(). these should be generic to "bid". likewise all the other functions